azure-pipelines.yml

```diff
@@ -1,342 +1,384 @@
+                  action: ${{ parameters.action }}
+                  terraformSubscriptionID: ${{ component.terraformSubscriptionID }}
+                  backendAzureRmResourceGroupName: ${{ component.backendAzureRmResourceGroupName }}
+                  setClusterNumbersEnvVar: true
+                  ${{ if ne(parameters['cluster'], 'All') }}:
+                    targetCommand: >
+                      -target azurerm_resource_group.kubernetes_resource_group["\"${{parameters.cluster}}\""]
+                      -target module.kubernetes["\"${{parameters.cluster}}\""]
+
+          - job: "DeployInfrastructuremis"
+            dependsOn: DeployInfrastructure
+            steps:
+              - template: pipeline-steps/deploy-service.yaml
+                parameters:
+                  environment: ${{ component.env }}
+                  location: ${{ parameters.location }}
+                  stack: "aks-mis"
+                  project: $(project)
+                  tfInitSub: ${{ variables.tfInitSub }}
+                  serviceConnection: ${{ component.serviceConnection }}
+                  action: ${{ parameters.action }}
+                  setClusterNumbersEnvVar: false
+                  terraformSubscriptionID: ${{ component.terraformSubscriptionID }}
+                  backendAzureRmResourceGroupName: ${{ component.backendAzureRmResourceGroupName }}
+                  ${{ if ne(parameters['cluster'], 'All') }}:
+                    targetCommand: >
+                      -target azurerm_role_assignment.uami_cft_rg_identity_operator["\"${{parameters.cluster}}\""]
+                      -target azurerm_role_assignment.preview_externaldns_read_rg["\"${{parameters.cluster}}\""]
+                      -target azurerm_role_assignment.preview_externaldns_dns_zone_contributor["\"${{parameters.cluster}}\""]
+                      -target azurerm_role_assignment.preview2aat_cft_rg_identity_operator["\"${{parameters.cluster}}\""]
+                      -target data.azurerm_kubernetes_cluster.kubernetes["\"${{parameters.cluster}}\""]
+
+      - stage: "BootStrapClusters_${{ component.env }}"
+        displayName: "${{ upper(component.env) }}: BootStrap Clusters"
+        dependsOn:
+          - "Aks_${{ component.env }}"
         variables:
-          clusters: ${{ parameters.cluster }}
-        steps:
-          - template: pipeline-steps/bootstrap.yaml
-            parameters:
-              environment: ${{ parameters.env }}
-              cluster: ${{ parameters.cluster }}
-              project: $(project)
-              serviceConnection: $(serviceConnection)
-              aksKeyVault: $(aksKeyVault)
-              terraformSubscriptionID : $(terraformSubscriptionID)
+          existing_clusters: $[ dependencies.CheckingClusters.outputs['CheckingClusters.cluster_deploy'] ]
+          cluster: ${{ parameters.cluster }}
+        jobs:
+          - job: BootStrap
+            condition: |
+              or(
```

Comment:
You can't mix the conditions here,
You need one at the Stage level for:
```
and(succeeded(), eq(variables['Build.Reason'], 'Manual'), eq(variables['existing_clusters'], variables['cluster'])),
and(succeeded(), eq(variables['isMain'], true), eq(variables['isAutoTriggered'], true)),
and(succeeded(), eq(variables['isAutoTriggered'], true), eq(variables['Build.Reason'], 'PullRequest'))
```

and then another at the job level matching what was already there 

```
and(succeeded(), eq('${{ parameters.action }}', 'apply')),
```

Because they use OR statements, the first condition on the stage says will check if its one of those options are validate the run type. The second condition will then validate is an apply.
This will ensure that only manual builds were the variables match can get to this stage and then it must be apply to run.

Your current method means it could be any of those one conditions to cause this to run.
Apply when 00 != 01 would be incorrect.